tests/tests.rs

```diff
@@ -39,24 +41,16 @@ proptest! {
     }
 }
 
-#[test]
-fn test_public_key_ed25519_to_string() {
-    // Valid account.
-    let r = Strkey::PublicKey(PublicKey([
-        0x36, 0x3e, 0xaa, 0x38, 0x67, 0x84, 0x1f, 0xba, 0xd0, 0xf4, 0xed, 0x88, 0xc7, 0x79, 0xe4,
-        0xfe, 0x66, 0xe5, 0x6a, 0x24, 0x70, 0xdc, 0x98, 0xc0, 0xec, 0x9c, 0x07, 0x3d, 0x05, 0xc7,
-        0xb1, 0x03,
-    ]))
-    .to_string();
-    assert_eq!(
-        r,
-        "GA3D5KRYM6CB7OWQ6TWYRR3Z4T7GNZLKERYNZGGA5SOAOPIFY6YQHES5",
-    );
-}
-
 proptest! {
     #[test]
     fn test_public_key_ed25519_to_string_doesnt_panic(data: [u8; 32]) {
         Strkey::PublicKey(PublicKey(data)).to_string();
     }
 }
+
+fn test_valid_strkey_roundtrip(str: &str, strkey: &Strkey) {
```

Comment:
Suggest using the term assert instead of test since this isn't a unit test, but a helper that makes some assertions. I'd avoid the var name `str` too since it is also a common type.
```suggestion
fn assert_convert_roundtrip(s: &str, strkey: &Strkey) {
```